src/getEditorNamespace.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import loadScript from 'load-script';
+
+/* global CKEDITOR */
+
+let promise;
+
+export default function getEditorNamespace( editorURL, namespaceLoaded ) {
+	if ( 'CKEDITOR' in window ) {
+		return Promise.resolve( CKEDITOR );
+	}
+
+	if ( editorURL.length < 1 ) {
+		throw new TypeError( 'CKEditor URL must be a non-empty string.' );
+	}
+
+	if ( !promise ) {
+		promise = getEditorNamespace.scriptLoader( editorURL ).then( res => {
+			promise = undefined;
```

Comment:
Take a look at the bigger picture:

https://github.com/ckeditor/ckeditor4-integrations-common/blob/3cc8e122f77b892d615d2bf0450612f8bada6621/src/getEditorNamespace.js#L10-L36

`promise` is pre-cached in module scope to make sure multiple calls to the function won't cause multiple script loads (e.g. when you are loading multiple components in Vue/React/Angular). Cache needs to be restored once the promise is fulfilled, otherwise, the same instance of the editor namespace will be cached for the whole app lifetime which maybe not what you are expecting. AFAIK, for now, it's just required to simplify testing. However, if you directly `delete window.CKEDITOR`, I suppose it's unexpected that the same instance will be returned again despite your intention.